geps/gep-1619/index.md

```diff
@@ -705,21 +665,23 @@ spec:
     - name: servicev2
       weight: 100
 ---
-kind: SessionPersistencePolicy
+kind: LoadBalancerPolicy
 metadata:
-  name: spp-split-route
+  name: lbp-split-route
 spec:
   targetRef:
-    kind: HTTPRoute
-    name: split-route
-  type: HTTPCookie
+    kind: Service
+    Name: servicev1
+  sessionPersistence:
+    name: split-route-cookie
 ```
 
 A potentially unexpected situation occurs when:
+
 1. Curl to `/a` which establishes a persistent session with `servicev1`
 2. Curl to `/b` routes to `servicev1` due to route persistence despite `weight: 0` configuration
 
-In this scenario, implementations should give precedence to session persistence, regardless of the `weight`
+In this scenario, implementations MUST give precedence to session persistence, regardless of the `weight`
```

Comment:
In this case both cookies may work semantically - if the first request goes to A, the cookieA is set and future requests use it. Same for B. However not sure if envoy allows this - so validation may reject it.

IMO we don't have very strong use cases for allowing users to customize the cookie names - and some managed implementations don't support it, so at most it should be optional or 'hint'.

Reply:
So you are saying if the client presents two cookies simultaneously in a traffic splitting scenario? I think it should just be the cookie that is listed first in the client's HTTP request. That seems the simplest. Is that what you meant by "semantically" @costinm?

If I understand this correctly, it's a bit unlikely of a scenario. If you provide a cookie, the `set-cookie` response header _shouldn't_ be set, and if it was set for some reason, it _shouldn't_ be set with a name & value to for another backend. Am I understanding correctly?

And yes, `sessionName` is implementation-specific (because it's not testable) support-level and optional.

Reply:
Yeah, I think you're right. I'm just trying to find edge cases when persistence config is set differently for traffic splitting backends. It seems like there are lots of possible rough edges when combining backend-attached session persistence with traffic splitting.